### PR TITLE
[CSL-996] add support for hidden facets

### PR DESCRIPTION
**Pull request by stanlp1**
Fixed tests and added support for hidden facets

### CODE REVIEW

File: AutocompleteClient/FW/Logic/Query/CIOQueryFmtOptions.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//  CIOQueryFmtOptions.swift
+//  AutocompleteClient
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+public typealias FmtOptions = (key: String, value: String)
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Query/CIOQueryFmtOptions.swift:11]: Should this be `FmtOption` (singular)? Same goes for the places where it's used below

**Review by jjl014 — COMMENTED** [review]
Lgtm! Good stuffs 😄 

Left one comment around the format options.

File: AutocompleteClient/FW/Logic/Query/CIOQueryFmtOptions.swift (hunk 1 above)
Comment (stanlp1) [reply AutocompleteClient/FW/Logic/Query/CIOQueryFmtOptions.swift]: yep

File: AutocompleteClient/FW/Logic/Request/CIOSearchQuery.swift [hunk 2]
```diff
@@ -72,14 +78,15 @@ public struct CIOSearchQuery: CIORequestData {
      let searchQuery = CIOSearchQuery(query: "red", filters: CIOQueryFilters(groupFilter: nil, facetFilters: facetFilters), page: 1, perPage: 30, section: "Products", hiddenFields: ["price_CA", "currency_CA"])
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOSearchQuery.swift:78]: Forgot to mention this in the previous review, but could we also update the usage examples as well? Thanks!

**Review by jjl014 — APPROVED** [review]
Lgtm! I did notice some other minor things that we might want to update as well as updating the usage examples. Otherwise, this looks great! 😄

**Review by crgee1 — APPROVED** [review]
ship it!

**Pull request merged**